contentctl/objects/detection_tags.py

```diff
@@ -52,6 +52,8 @@ def risk_score(self) -> int:
 
     mitre_attack_id: List[MITRE_ATTACK_ID_TYPE] = []
     nist: list[NistCategory] = []
+
+    # TODO (#249): Add pydantic validator to ensure observables are unique within a detection
```

Comment:
Nearly all of our `list[ALMOST_ANY_TYPE]` typing should exhibit this uniqueness constraint. We have a few options here:
1. Convert as many things as possible to sets (`SecurityContentObject_Abstract` has the appropriate methods for object comparisons, but we have list[] of things that are not `SecurityContentObject_Abstract` as well). For example, a detection should not be able to have the same AnalyticStory twice.  However, the issue here is we would want duplicates to throw an Exception, not just appear once in the results. Additionally, the orders of some of our serialization are important (so we would need to make sure when they are written to output files, the values are sorted).
2. Define a custom type, like `listWithNoDuplicates[ANY_TYPE]` which ensures no duplicates exist during validation and throws a ValueError on the field if they do.
3. Something else I haven't thought of.

This is out of scope for this PR, but a good point to discuss.

Reply:
Agreed w/ converting as many things to sets as possible

Re: ensuring duplicates are not added to the set, I would just define a validator as part of our base model and apply it to each field we want it on directly